src/EFCore.PG/Query/ExpressionTranslators/Internal/NpgsqlArrayMethodCallTranslator.cs

```diff
@@ -0,0 +1,401 @@
+        /// <param name="expression">The expression to translate.</param>
+        /// <returns>
+        /// A translated expression or null if no translation is supported.
+        /// </returns>
+        [CanBeNull]
+        Expression ListInstanceHandler([NotNull] MethodCallExpression expression)
+        {
+            if (!(expression.Object is Expression instance))
+                return null;
+
+            if (!instance.Type.IsGenericType || instance.Type.GetGenericTypeDefinition() != typeof(List<>))
+                return null;
+
+            switch (expression.Method.Name)
+            {
+            case "get_Item":
+                return MakeIndex(instance, expression.Arguments[0]);
+
+            case nameof(IList.IndexOf) when VersionAtLeast(9, 5):
+                return Coalesce(new SqlFunctionExpression("array_position", typeof(int), new[] { instance, expression.Arguments[0] }));
+
+            default:
+                return null;
+            }
+        }
+
+        /// <summary>
+        /// Translates instance methods defined on <see cref="string"/>.
+        /// </summary>
+        /// <param name="expression">The expression to translate.</param>
+        /// <returns>
+        /// A translated expression or null if no translation is supported.
+        /// </returns>
+        [CanBeNull]
+        static Expression StringInstanceHandler([NotNull] MethodCallExpression expression)
+        {
+            if (!(expression.Object is Expression instance))
+                return null;
+
+            if (instance.Type != typeof(string))
+                return null;
+
+            switch (expression.Method.Name)
+            {
+            case "get_Chars":
+                return MakeIndex(instance, expression.Arguments[0]);
+
+            default:
+                return null;
+            }
+        }
+
+        #endregion
+
+        #region Helpers
+
+        /// <summary>
+        /// True if <see cref="_postgresVersion"/> is null, greater than, or equal to the specified version.
+        /// </summary>
+        /// <param name="major">The major version.</param>
+        /// <param name="minor">The minor version.</param>
+        /// <returns>
+        /// True if <see cref="_postgresVersion"/> is null, greater than, or equal to the specified version.
+        /// </returns>
+        bool VersionAtLeast(int major, int minor)
+            => _postgresVersion is null || new Version(major, minor) <= _postgresVersion;
+
+        /// <summary>
+        /// Creates an <see cref="IndexExpression"/> from the <paramref name="instance"/> and the <paramref name="index"/>.
+        /// </summary>
+        /// <param name="instance">The instance to index.</param>
+        /// <param name="index">The index value.</param>
+        /// <returns>
+        /// An <see cref="IndexExpression"/>.
+        /// </returns>
+        [NotNull]
+        static IndexExpression MakeIndex([NotNull] Expression instance, [NotNull] Expression index)
```

Comment:
Maybe add to the documentation that this is not used for arrays, only lists.

Reply:
It's also used for string indexers, but I'll document it all the same.